### PR TITLE
Parallel reconst workflows

### PR DESCRIPTION
**Pull request by sitek**
Adding parallelization to `peaks_from_models` for CSA and CSD in the reconstruction workflows.

### CODE REVIEW

**Comment by arokem** [comment]
Do we have parallelization implemented in any other workflows? I think we could simplify a bit by using just one input parameter: `nbr_processes` that defaults to False. When this is set to a number > 0, that would be the number of processes to spawn, if set to -1, that means use all cpus. If it's not set, use the  default (no parallelization). Would that be consistent with other workflows?

File: dipy/workflows/reconst.py [hunk 1]
```diff
@@ -440,6 +440,12 @@ def run(self, input_files, bvalues_files, bvectors_files, mask_files,
         odf_to_sh_order : int, optional
             Spherical harmonics order used for peak_from_model to compress
             the ODF to spherical harmonics coefficients (default 8)
+        parallel : bool, optional
+            Whether to use parallelization in peak-finding during the
+            calibration procedure. Default: True
+        nbr_processes: int
```
[the remark this answers is not in the document]
Comment (sitek) [reply dipy/workflows/reconst.py]: done

File: dipy/workflows/reconst.py [hunk 2]
```diff
@@ -600,6 +607,12 @@ def run(self, input_files, bvalues_files, bvectors_files, mask_files,
             Threshold used so that norm(bvec)=1 (default 0.01)
         extract_pam_values : bool, optional
             Wheter or not to save pam volumes as single nifti files.
+        parallel : bool, optional
+            Whether to use parallelization in peak-finding during the
+            calibration procedure. Default: True
+        nbr_processes: int
```
[the remark this answers is not in the document]
Comment (sitek) [reply dipy/workflows/reconst.py]: done

**Comment by skoudoro** [comment]
Hi @sitek,

Can you add a test on `dipy/workflow/reconst`? 

After that, this PR it ready to go.

> Do we have parallelization implemented in any other workflows?

No, and I think, we should add a global variable for all workflows (like `--force`). This need to be implemented in a new PR.

> When this is set to a number > 0, that would be the number of processes to spawn, if set to -1, that means use all cpus. If it's not set, use the default (no parallelization). Would that be consistent with other workflows?

I agree with this point but we need a specific PR and check all DIPY codebase. Currently we are not consistent and I think it is a bad idea to update it on this PR. What do you think @arokem?

**Comment by sitek** [comment]
> Hi @sitek,
> 
> Can you add a test on `dipy/workflow/reconst`?
> 
> After that, this PR it ready to go.

Added (and fixed docs)

**Comment by arokem** [comment]
@skoudoro : we need to make sure that we are consistent in how we define these things, but I agree that this is a matter for a different day... 

I think that this PR is ready to go! Will merge in a couple of days, unless anyone objects.

**Comment by arokem** [comment]
CI failure is unrelated. Merging.

**Pull request merged**